provider/packet/environ.go

```diff
@@ -0,0 +1,519 @@
+		Tags:         packetTags,
+	}
+
+	d, _, err := e.packetClient.Devices.Create(device)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	d, err = waitDeviceActive(e.packetClient, d.ID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	inst := &packetDevice{e, d}
+	amd64 := arch.AMD64
+	mem, err := strconv.ParseUint(d.Plan.Specs.Memory.Total[:len(d.Plan.Specs.Memory.Total)-2], 10, 64)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var cpus uint64
+	if len(inst.Plan.Specs.Cpus) > 0 {
+		cpus = uint64(inst.Plan.Specs.Cpus[0].Count)
+	}
+
+	hc := &instance.HardwareCharacteristics{
+		Arch: &amd64,
+		Mem:  &mem,
+		// RootDisk: &instanceSpec.InstanceType.RootDisk,
+		CpuCores: &cpus,
+	}
+
+	return &environs.StartInstanceResult{
+		Instance: inst,
+		Hardware: hc,
+	}, nil
+}
+
+// supportedInstanceTypes returns the instance types supported by packet.
+func (e *environ) supportedInstanceTypes() ([]instances.InstanceType, error) {
+	plans, _, err := e.packetClient.Plans.List(new(packngo.ListOptions))
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving supported instance types")
+	}
+
+	var instTypes []instances.InstanceType
+nextPlan:
+	for _, plan := range plans {
+		if !validPlan(plan) {
+			continue
+		}
+
+		// TODO: Only work with this plan type for now to avoid
+		// spinning up more expensive instances while testing.
+		if plan.Name != "t1.small.x86" {
+			continue
+		}
+
+		var instArch string
+		switch {
+		case strings.HasSuffix(plan.Name, ".x86"):
+			instArch = arch.AMD64
+		case strings.HasSuffix(plan.Name, ".arm"):
+			instArch = arch.ARM64
+		default:
+			continue nextPlan
+
+		}
+
+		mem, err := parseMemValue(plan.Specs.Memory.Total)
+		if err != nil {
+			continue
+		}
+
+		instTypes = append(instTypes,
+			instances.InstanceType{
+				Id:       plan.ID,
+				Name:     plan.Name,
+				CpuCores: uint64(plan.Specs.Cpus[0].Count),
+				Mem:      mem,
+				Arches:   []string{instArch},
+				// Scale per hour costs so they can be represented as an integer for sorting purposes.
+				Cost: uint64(plan.Pricing.Hour * 1000.0),
+				// TODO: returned by packet's API but not exposed by the packngo client
+				// Deprecated: plan.Legacy,
+			})
+	}
+
+	return instTypes, nil
+}
+
+func validPlan(plan packngo.Plan) bool {
+	isInvalid := plan.Pricing == nil ||
+		plan.Specs == nil ||
+		plan.Specs.Memory == nil ||
+		len(plan.Specs.Cpus) == 0 || plan.Specs.Cpus[0].Count == 0
+
+	return !isInvalid
+}
+
+func parseMemValue(v string) (uint64, error) {
+	var scaler = uint64(1)
+	if strings.HasSuffix(v, "GB") {
+		scaler = 1024
+		v = strings.TrimSuffix(v, "GB")
+	}
+
+	val, err := strconv.ParseUint(v, 10, 64)
+	return val * scaler, err
+}
+
+func (e *environ) findInstanceSpec(controller bool, allImages []*imagemetadata.ImageMetadata, instanceTypes []instances.InstanceType, ic *instances.InstanceConstraint) (*instances.InstanceSpec, error) {
+	// TODO: remove this when we image discovery via streams works
```

Comment:
@achilleasa we had a conversation about this yesterday. Should `allImages` parameter be mapped according to Equinix standards or was it something else?

Reply:
If Equinix does not mirror the images specified by the streams I think you will need to map the image name to something that the packet API can work with. Alternatively, if the packet API supports listing of available images in the target region, you might be able to use it to generate your own metadata which you can then pass through to the filtering code below.